TypeCobol/Compiler/Nodes/Data.cs

```diff
@@ -317,27 +337,34 @@ public virtual long PhysicalLength
                     {
                         _physicalLength = GetPhysicalLength();
                     }
+                    else if (((DataDefinitionEntry) CodeElement).LevelNumber?.Value == 88)
+                    {
+                        //Exception case if this is a level 88
+                        _physicalLength = 0;
+                    }
+                    else if (DataType == DataType.Boolean)
+                    {
+                        //exception case if this is a type bool
+                        _physicalLength = 1;
+                    }
                     else if (TypeDefinition != null)
                     {
                         _physicalLength = TypeDefinition.PhysicalLength;
                     }
                     else
                     {
+                        if (_physicalLength == -1)
+                            _physicalLength = 0;
+
                         foreach (var node in children)
                         {
                             var dataDefinition = (DataDefinition)node;
-                            if (dataDefinition != null)
+                            
+                            if (dataDefinition is DataRedefines == false)
                             {
-                                _physicalLength += dataDefinition.PhysicalLength;
-                                _physicalLength += dataDefinition.SlackBytes;
-                            }
-
-                            if (dataDefinition is DataRedefines)
-                            {
-                                SymbolReference redefined = ((DataRedefinesEntry)dataDefinition.CodeElement)?.RedefinesDataName;
-                                var result = SymbolTable.GetRedefinedVariable((DataRedefines) dataDefinition, redefined);
-
-                                _physicalLength -= result.PhysicalLength > dataDefinition.PhysicalLength ? dataDefinition.PhysicalLength : result.PhysicalLength;
+                                var redefines = dataDefinition.GetBiggestRedefines();
+                                _physicalLength += Math.Max(redefines?.PhysicalLength ?? 0, dataDefinition.PhysicalLength);
+                                _physicalLength += redefines?.SlackBytes ?? dataDefinition.SlackBytes;
```

Comment:
Is it ok if we take:
`dataDefinition.PhysicalLength` (because its the max)
then `redefines?.SlackBytes` ?

It seems possible to happen with this code.

Reply:
Actually it is not, that's my bad, I didn't fully understood when we can use a REDEFINES. The latest commit fix this issue